marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/IOEndpointImpl.java

```diff
@@ -119,21 +123,23 @@ public void checkAllowedArgs(InputStream endpointState, SessionState session, In
         private byte[]         endpointState;
         private byte[]         workUnit;
         private SessionState   session;
+        private CallContextImpl callContext;
 
         private long           callCount = 0;
 
         BulkIOEndpointCallerImpl(IOEndpointImpl endpoint) {
             if (endpoint == null)
                 throw new IllegalArgumentException("null endpoint definition");
+            this.callContext = new CallContextImpl(endpoint);
             this.endpoint = endpoint;
         }
 
         private IOEndpointImpl getEndpoint() {
-            return this.endpoint;
+            return callContext.getEndpoint();
```

Comment:
See the comment below about the CallContext constructor.  I'm wondering whether it might be beneficial to access the endpoint in the field instead of in the call context.